README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->
<!--[![Travis-CI Build Status](https://travis-ci.org/SMAC-Group/imudata.svg?branch=master)](https://travis-ci.org/SMAC-Group/imudata)-->
[![Project Status: Active](http://www.repostatus.org/badges/latest/active.svg)](http://www.repostatus.org/#active) [![Licence](https://img.shields.io/badge/licence-CC%20BY--NC--SA%204.0-blue.svg)](https://www.gnu.org/licenses/gpl-3.0.en.html) [![minimal R version](https://img.shields.io/badge/R%3E%3D-3.4.0-6666ff.svg)](https://cran.r-project.org/) [![packageversion](https://img.shields.io/badge/Package%20version-0.1.0-orange.svg?style=flat-square)](commits/develop) [![Last-changedate](https://img.shields.io/badge/last%20change-2018--02--11-yellowgreen.svg)](/commits/master)

`imudata` Overview <a href="https://smac-group.com/"><img src="man/figures/logo.png" align="right" style="width: 20%; height: 20%"/></a>
========================================================================================================================================

This package is meant to serve as a data collection tool for IMU data. This data can be used as a means to assess and test methods designed to analyse IMU error signals (i.e. long and complex autocorrelated signals). An example method used for this kind of data is implemented in the `gmwm` R package which can also model the latent models that often characterize this data.

The datasets available within the package are:

-   `imu6` - A MTi-G MEMS-IMU dataset with 6 columns corresponding to the stochastic error measurements coming from the X, Y and Z axes for gyroscopes and accelerometers respectively and taken from a sensor calibrated in stationary conditions for 2 hours at 100Hz.
-   `cont.imu1` - A MTi-G MEMS-IMU dataset with 1 column corresponding to the stochastic error measurements coming from the X-axis gyroscope and taken from a sensor calibrated in stationary conditions for 2 hours at 100Hz and possibly suffering from some contaminated measurements (e.g. outliers).
-   `navchip` - A Navchip MEMS-IMU dataset with 6 columns, Axis: X,Y,Z - Type: Gyroscope & Accelerometer, from a stationary sensor run for 4 Hours
-   `imar.gyro` - An IMAR MEMS-IMU dataset with 3 columns, Axis: X,Y,Z - Type: Accelerometer, from a stationary sensor run for 4 Hours
-   `ln200.gyro` - A LN200 MEMS-IMU dataset with 3 columns, Axis: X,Y,Z - Type: Gyroscope, from a stationary sensor run for 6 Hours
-   `ln200.accel` - A LN200 MEMS-IMU dataset with 3 columns, Axis: X,Y,Z - Type: Accelerometer, from a stationary sensor run for 6 Hours
-   ADD MISSING DATASETS

The first 6 datasets can be used as examples for the functions `auto.imu()`, `gmwm.imu()`, `wvar.imu()`, and `imu()` of the `gmwm` R package. Note that the `cont.imu1` data can be used as an illustration of the robustness properties of the robust version the Generalized Method of Wavelet Moments (GMWM). Here is a simple example:


![](https://raw.githubusercontent.com/smac-group/imudata/master/img/demo.gif)

Install Instructions
====================

To install the package you can use:

``` r
# Install R devtools
install.packages("devtools")

# Install the package from github
devtools::install_github("SMAC-Group/imudata")
```

Licensing
=========

The license this source code is released under is the Creative Commons Attribution NonCommercial ShareAlike (CC-NC-SA). In some cases, the GPL license does apply. However, in the majority of the cases, the license in effect is the Creative Commons Attribution NonCommercial ShareAlike (CC-NC-SA) as the computational code is heavily dependent on armadilllo, which has an MIT license that enables us to recast our code to the Creative Commons Attribution NonCommercial ShareAlike (CC-NC-SA). See the LICENSE file for full text. Otherwise, please consult [TLDR Legal](https://tldrlegal.com/license/creative-commons-attribution-noncommercial-sharealike-(cc-nc-sa)) or [CC](https://creativecommons.org/licenses/by-nc-sa/4.0/#) which will provide a synopsis of the restrictions placed upon the data and code.


NEWS.md
# imudata 0.2.0

## New

* `kvh1750.acc` - Six samples collected from KVH1750 IMU at 100Hrz Axis: X,Y,Z - Type: Acce,  from a stationary sensor run for 3 hours
* `kvh1750.gyro` - Six samples collected from KVH1750 IMU at 100Hrz Axis: X,Y,Z - Type: Accelerometer,  from a stationary sensor run for 3 hours
* `mtig1khrz` - Six samples collected from MTI-G-710 IMU at 1000 Hz Axis: X,Y,Z - Type: Gyroscope & Accelerometer,  from a stationary sensor run for 10 minutes
* `adis16405` - Six samples collected from ADIS 16405 IMU at 100Hrz Axis: X,Y,Z - Type: Gyroscope & Accelerometer,  from a stationary sensor run for 3 hours



R/imudata.R
#' Inertial Measurement Units (IMU) Data Package
#'
#' This package is meant to serve as a data collection tool for IMU data. 
#' This data can be used as a means to assess and test methods designed to analyse 
#' IMU error signals (i.e. long and complex autocorrelated signals). An example method
#' used for this kind of data is implemented in the GMWM R package which can also model 
#' the latent models that often characterize this data.
#' 
#' @details
#' \tabular{ll}{
#' Package: \tab gmwmdata\cr
#' Type: \tab Package\cr
#' Version: \tab 3.0.0\cr
#' Date: \tab 2018-02-10\cr
#' License: \tab CC BY-NC-SA 4.0\cr
#' }
#'
#' @author
#' Stephane Guerrier \email{stephane@@psu.edu},
#' Roberto Molinari \email{roberto.molinari@hotmail.it},
#' James Balamuta \email{balamut2@@illinois.edu},
#' Geatan Bakalli \email{Geatan.Bakalli@unige.ch},
#' Ahmed Radi \email{ahmed.elboraee@ucalgary.ca},
#' Wenchao Yang \email{wyang40@@illinois.edu}
#'
#' Stephane Guerrier \email{stephane@@psu.edu}
#' @docType package
#' @name imudata
#' @exportPattern ^[[:alpha:]]+
#' @seealso \code{\link[imudata]{cont.imu1}}, \code{\link[imudata]{imu6}}, \code{\link[imudata]{imar.gyro}}, \code{\link[imudata]{ln200.accel}}, \code{\link[imudata]{ln200.gyro}}, \code{\link[imudata]{mtig1khrz}}, \code{\link[imudata]{navchip}}, \code{\link[imudata]{adis16405}},  \code{\link[imudata]{kvh1750.gyro}} and \code{\link[imudata]{kvh1750.accel}}
#' @examples
#' # Please see the datasets
NULL


data-raw/process_imu.R
library(gmwm)
library(imudata) # install via install_imudata()

# Change this to your BoxSync path *e.g. BoxSync => Box Sync
loc = "/Users/james/BoxSync/GMWM_project/IMU_data/sample_data"

# 4 Hours IMAR
im = read.imu(paste0(loc,"/imar_4h.imu"),'IMAR')

imar.gyro = im[,1:3]

save(imar.gyro, file="data/imar.gyro.rda", compress="xz")


# 6 Hours LN200
ln200_6h = read.imu(paste0(loc,"/ln200_6h.imu"),'LN200')

ln200.gyro = ln200_6h[,1:3]

ln200.accel = ln200_6h[,4:6]

save(ln200.gyro, file="data/ln200.gyro.rda", compress="xz")
save(ln200.accel, file="data/ln200.accel.rda", compress="xz")


# 4 Hours Navchip
navchip = read.imu(paste0(loc,"/navchip_4h.imu"),'NAVCHIP_FLT')

save(navchip, file="data/navchip.rda", compress="xz")


# Convert IMU6
data(imu6)
mtig = imu(imu6, gyros = 1:3, accels = 4:6)

save(mtig, file="data/mtig.rda", compress="xz")

# Convert CONT.IMU1
data(cont.imu1)
cont.imu1 = imu(cont.imu1, gyros = 1)
save(cont.imu1, file="data/cont.imu1.rda", compress="xz")

# Convert KVH
data(kvh)
kvh = KVH1750imuAcc
save(kvh, file="data/kvh.rda", compress="xz")


R/data.R
#' @title IMU Data from a XSens MTi-G sensor
#' @description This data set contains gyroscope and accelerometer data from a XSens MTi-G sensor.
#' @format A \code{matrix} with 873,684 observations and 6 columns. The columns are defined as follows:
#' \describe{
#'  \item{\code{Gyro. X}}{Gyroscope X-Axis}
#'  \item{\code{Gyro. Y}}{Gyroscope Y-Axis}
#'  \item{\code{Gyro. Z}}{Gyroscope Z-Axis}
#'  \item{\code{Accel. X}}{Accelerometer X-Axis}
#'  \item{\code{Accel. Y}}{Accelerometer Y-Axis}
#'  \item{\code{Accel. Z}}{Accelerometer Z-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @details 
#' This data set is equivalent to the \code{\link[imudata]{mtig}} data set. The only difference
#' is that this data set acts as a way to showcase the \code{\link[gmwm]{imu}} cast function whereas 
#' \code{\link[imudata]{mtig}} has already been cast using \code{\link[gmwm]{imu}}.
#' @seealso \code{\link[imudata]{mtig}}
#' @examples
#' \dontrun{
#' data(imu6)
#' summary(imu6)
#' }
"imu6"



#' @title IMU Data over 4 Hours from a navchip sensor
#' @description This data set contains gyroscope and accelerometer data from navchip-based sensors.
#' @format An \code{imu} with 3,105,250 observations and 6 sensors. The sensors are defined as follows:
#' \describe{
#'  \item{\code{Gyro. X}}{Gyroscope X-Axis}
#'  \item{\code{Gyro. Y}}{Gyroscope Y-Axis}
#'  \item{\code{Gyro. Z}}{Gyroscope Z-Axis}
#'  \item{\code{Accel. X}}{Accelerometer X-Axis}
#'  \item{\code{Accel. Y}}{Accelerometer Y-Axis}
#'  \item{\code{Accel. Z}}{Accelerometer Z-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @examples
#' \dontrun{
#' data(navchip)
#' navchip
#' }
"navchip"

#' @title IMU Data over 4 Hours from IMAR Gyroscopes
#' @description This data set contains gyroscope from multiple IMAR Gyroscopes.
#' @format An \code{imu} with 5,760,000 observations and 3 sensors. The sensors are defined as follows:
#' \describe{
#'  \item{\code{Gyro. X}}{Gyroscope X-Axis}
#'  \item{\code{Gyro. Y}}{Gyroscope Y-Axis}
#'  \item{\code{Gyro. Z}}{Gyroscope Z-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @examples
#' \dontrun{
#' data(imar.gyro)
#' imar.gyro
#' }
"imar.gyro"


#' @title IMU Data over 6 Hours from LN200 Gyroscopes
#' @description This data set contains gyroscope from multiple LN200 Gyroscopes.
#' @format An \code{imu} with 8,640,000 observations and 3 sensors. The sensors are defined as follows:
#' \describe{
#'  \item{\code{Gyro. X}}{Gyroscope X-Axis}
#'  \item{\code{Gyro. Y}}{Gyroscope Y-Axis}
#'  \item{\code{Gyro. Z}}{Gyroscope Z-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @examples
#' \dontrun{
#' data(ln200.gyro)
#' ln200.gyro
#' }
"ln200.gyro"

#' @title IMU Data over 6 Hours from LN200 Accelerometers
#' @description This data set contains gyroscope from multiple LN200 Accelerometers
#' @format An \code{imu} with 8,640,000 observations and 3 sensors. The sensors are defined as follows:
#' \describe{
#'  \item{\code{Accel. X}}{Accelerometer X-Axis}
#'  \item{\code{Accel. Y}}{Accelerometer Y-Axis}
#'  \item{\code{Accel. Z}}{Accelerometer Z-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @examples
#' \dontrun{
#' data(ln200.accel)
#' ln200.accel
#' }
"ln200.accel"

#' @title Containment IMU Data from a XSens MTi-G sensor
#' @description This data set contains gyroscope data
#' @format An \code{imu} with 500,000 observations and 1 variable. The variable is as follows:
#' \describe{
#'  \item{\code{Gyro. X}}{Gyroscope X-Axis}
#' }
#' @source Geodetic Engineering Laboratory (TOPO), Swiss Federal Institute of Technology Lausanne (EPFL)
#' @author JJB
#' @examples
#' \dontrun{
#' data(cont.imu1)
#' cont.imu1
#' }
"cont.imu1"


#' @title Six samples collected from KVH1750 IMU at 100Hrz (only accelerometers) over periods of about 3 hours
#' @description This data set contains six samples collected from KVH1750 IMU. 
#' @format An \code{mimu} from object
#' @source Department of Geomatics Engineering, University of Calgary
#' @author Stéphane Guerrier, Gaetan Bakalli
#' @examples
#' \dontrun{
#' data(kvh1750.acc)
#' kvh1750.acc
#' }
"kvh1750.acc"

#' @title Six samples collected from KVH1750 IMU at 100Hrz (only accelerometers) over periods of about 3 hours
#' @description This data set contains six samples collected from KVH1750 IMU. 
#' @format An \code{mimu} from object
#' @source Department of Geomatics Engineering, University of Calgary
#' @author Stéphane Guerrier, Gaetan Bakalli
#' @examples
#' \dontrun{
#' data(kvh1750.gyro)
#' kvh1750.gyro
#' }
"kvh1750.gyro"


#' @title Six samples collected from MTI-G-710 IMU at 1000 Hz over periods of about 10 minutes
#' @description This data set contains six samples collected from MTI-G-710 IMU at 1000 Hz. 
#' @format An \code{mimu} from object
#' @source Department of Geomatics Engineering, University of Calgary
#' @author Stéphane Guerrier, Gaetan Bakalli
#' @examples
#' \dontrun{
#' data(mtig1khrz)
#' mtig1khrz
#' }
"mtig1khrz"



#' @title Six samples collected from ADIS 16405 IMU at 100 Hz  over periods of about 3 hours
#' @description This data set contains six samples collected from ADIS 16405 IMU at 100 Hz. 
#' @format An \code{mimu} from object
#' @source Department of Geomatics Engineering, University of Calgary
#' @author Stéphane Guerrier, Gaetan Bakalli
#' @examples
#' \dontrun{
#' data(adis16405)
#' adis16405
#' }
"adis16405"





